// angular-feather/esm2015/feather/feather.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef, Input, Inject, ChangeDetectorRef } from '@angular/core';
import { Icons } from './icons.provider';
import { uppercamelcase } from './utils';
export class FeatherComponent {
    /**
     * @param {?} elem
     * @param {?} changeDetector
     * @param {?} icons
     */
    constructor(elem, changeDetector, icons) {
        this.elem = elem;
        this.changeDetector = changeDetector;
        this.icons = icons;
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    ngOnChanges(changes) {
        // icons are provided as an array of objects because of "multi: true"
        /** @type {?} */
        const icons = Object.assign({}, ...((/** @type {?} */ ((/** @type {?} */ (this.icons))))));
        /** @type {?} */
        const svg = icons[uppercamelcase(changes.name.currentValue)] || '';
        if (!svg) {
            console.warn(`Icon not found: ${changes.name.currentValue}\n` +
                `Refer to documentation on https://github.com/michaelbazos/angular-feather`);
        }
        this.elem.nativeElement.innerHTML = svg;
        this.changeDetector.markForCheck();
    }
}
FeatherComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'i-feather, feather-icon',
                template: "<ng-content></ng-content>\n",
                styles: [":host{display:inline-block;width:24px;height:24px;fill:none;stroke:currentColor;stroke-width:2px;stroke-linecap:round;stroke-linejoin:round}"]
            }] }
];
/** @nocollapse */
FeatherComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: ChangeDetectorRef },
    { type: Icons, decorators: [{ type: Inject, args: [Icons,] }] }
];
FeatherComponent.propDecorators = {
    name: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    FeatherComponent.prototype.name;
    /**
     * @type {?}
     * @private
     */
    FeatherComponent.prototype.elem;
    /**
     * @type {?}
     * @private
     */
    FeatherComponent.prototype.changeDetector;
    /**
     * @type {?}
     * @private
     */
    FeatherComponent.prototype.icons;
}
