// angular-feather/esm5/feather/feather.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, ElementRef, Input, Inject, ChangeDetectorRef } from '@angular/core';
import { Icons } from './icons.provider';
import { uppercamelcase } from './utils';
var FeatherComponent = /** @class */ (function () {
    function FeatherComponent(elem, changeDetector, icons) {
        this.elem = elem;
        this.changeDetector = changeDetector;
        this.icons = icons;
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    FeatherComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        // icons are provided as an array of objects because of "multi: true"
        /** @type {?} */
        var icons = Object.assign.apply(Object, tslib_1.__spread([{}], ((/** @type {?} */ ((/** @type {?} */ (this.icons)))))));
        /** @type {?} */
        var svg = icons[uppercamelcase(changes.name.currentValue)] || '';
        if (!svg) {
            console.warn("Icon not found: " + changes.name.currentValue + "\n" +
                "Refer to documentation on https://github.com/michaelbazos/angular-feather");
        }
        this.elem.nativeElement.innerHTML = svg;
        this.changeDetector.markForCheck();
    };
    FeatherComponent.decorators = [
        { type: Component, args: [{
                    // tslint:disable-next-line:component-selector
                    selector: 'i-feather, feather-icon',
                    template: "<ng-content></ng-content>\n",
                    styles: [":host{display:inline-block;width:24px;height:24px;fill:none;stroke:currentColor;stroke-width:2px;stroke-linecap:round;stroke-linejoin:round}"]
                }] }
    ];
    /** @nocollapse */
    FeatherComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: ChangeDetectorRef },
        { type: Icons, decorators: [{ type: Inject, args: [Icons,] }] }
    ]; };
    FeatherComponent.propDecorators = {
        name: [{ type: Input }]
    };
    return FeatherComponent;
}());
export { FeatherComponent };
if (false) {
    /** @type {?} */
    FeatherComponent.prototype.name;
    /**
     * @type {?}
     * @private
     */
    FeatherComponent.prototype.elem;
    /**
     * @type {?}
     * @private
     */
    FeatherComponent.prototype.changeDetector;
    /**
     * @type {?}
     * @private
     */
    FeatherComponent.prototype.icons;
}
