// angular-feather/esm5/feather/feather.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule, Optional } from '@angular/core';
import { FeatherComponent } from './feather.component';
import { Icons } from './icons.provider';
var FeatherModule = /** @class */ (function () {
    function FeatherModule(icons) {
        this.icons = icons;
        if (!this.icons) {
            throw new Error("No icon provided. Make sure to use 'FeatherModule.pick({ ... })' when importing the module\n" +
                "Refer to documentation on https://github.com/michaelbazos/angular-feather");
        }
    }
    /**
     * @param {?} icons
     * @return {?}
     */
    FeatherModule.pick = /**
     * @param {?} icons
     * @return {?}
     */
    function (icons) {
        return {
            ngModule: FeatherModule,
            providers: [
                { provide: Icons, multi: true, useValue: icons }
            ]
        };
    };
    FeatherModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [
                        FeatherComponent
                    ],
                    exports: [
                        FeatherComponent
                    ]
                },] }
    ];
    /** @nocollapse */
    FeatherModule.ctorParameters = function () { return [
        { type: Icons, decorators: [{ type: Optional }] }
    ]; };
    return FeatherModule;
}());
export { FeatherModule };
if (false) {
    /**
     * @type {?}
     * @private
     */
    FeatherModule.prototype.icons;
}
